src/js/components/sharedComponents/sidebar/SidebarLink.jsx

```diff
@@ -3,69 +3,55 @@
  * Created by Kevin Li 6/8/17
  */
 
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
-import { withRouter, Link } from 'react-router-dom';
+import { Link, useLocation } from 'react-router-dom';
 
 const propTypes = {
     section: PropTypes.string,
     label: PropTypes.string,
     active: PropTypes.string,
     onClick: PropTypes.func,
-    location: PropTypes.object,
     overLine: PropTypes.string
 };
 
-export class SidebarLink extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            url: ''
-        };
-
-        this.clickedLink = this.clickedLink.bind(this);
-    }
-
-    componentDidMount() {
-        this.prepareLink();
-    }
-
-    prepareLink() {
+const SidebarLink = (props) => {
+    const [url, setUrl] = useState('');
+    const location = useLocation();
+    const prepareLink = () => {
     // the URL base should be the current route
-        const currentRoute = this.props.location.pathname;
+        const currentRoute = location.pathname;
         // append the section as a query param
-        const url = `${currentRoute}?section=${this.props.section}`;
+        const tempUrl = `${currentRoute}?section=${props.section}`;
+        setUrl(tempUrl);
+    };
 
-        this.setState({
-            url
-        });
-    }
+    useEffect(() => {
+        prepareLink();
+    }, [prepareLink]);
```

Comment:
This dep isn't needed, I'm assuming it was auto-added, so we should use
// eslint-disable-next-line react-hooks/exhaustive-deps
here

Reply:
I'll change it but it doesn't make a difference.